benchmark/utils/timer.hpp

```diff
@@ -260,6 +264,7 @@ class CudaTimer : public Timer {
 protected:
     void tic_impl() override
     {
+        exec_->synchronize();
```

Comment:
For the `tic` it doesn't matter right? This is not timed because it's before the `eventRecord` and it could even help in ensuring nothing is running on the GPU when we start running things, like if there was a copy previously (for example, `x_clone`).

Reply:
the event should also be after the memcpy if it involves gpu.
For me, ensuring nothing is running on the GPU is the stuff before calling timer, but it does not hurt the timing step.